src/middlewared/middlewared/plugins/smb.py

```diff
@@ -1388,6 +1390,8 @@ async def pool_post_import(middleware, pool):
         By the time the post-import hook is called, the smb.configure should have
         already completed and initialized the SMB service.
         """
+        await middleware.call('sharing.smb.disable_acl_if_trivial')
```

Comment:
Is this always disabling ACL if the ACL on share is trivial? If so, this is wrong.

Reply:
This method only processes migrated shares on first corresponding pool import / dataset unlock and is no-op at other times.

Reply:
Good. Just verifying.